### ISSUE TITLE
Some materialized views not appearing

### ISSUE THREAD

**Issue by georgelroberts** [metabase/metabase#29784]
I am unable to see some materialized views from the data list when creating a question. This only seems to be happening to materialized views I've created recently, because I can see ones I created a few months ago.
<img width="450" alt="Screenshot 2023-04-04 at 09 47 25" src="https://user-images.githubusercontent.com/7758824/229724006-39671f8e-7b18-42d7-83be-d20d67952d29.png">

I am able to query it through SQL, but then of course can't apply a field filter to it.
<img width="303" alt="Screenshot 2023-04-04 at 09 46 50" src="https://user-images.githubusercontent.com/7758824/229724013-bf437f20-d580-4546-98d3-0a3e6f3b1424.png">

In admin settings I have tried re-scanning field values and syncing the database schema, but neither have worked (it has been around 2 weeks since I created a materialized view).

Troubleshooting info:
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.18+10",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.18",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.18+10",
    "os.name": "Linux",
    "os.version": "4.15.0-191-generic",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "mysql",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "13.8"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.5.0"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2023-01-27",
      "tag": "v0.45.2.1",
      "branch": "release-x.45.2.x",
      "hash": "1a59de7"
    },
    "settings": {
      "report-timezone": null
    }
  }
}

This is pretty severe for us, as it is blocking us creating a dashboard for clients.

**Comment by paoliniluis** [comment]
Have you done a sync? please try doing a sync and post the logs here so we can see what might be going on. Also please upgrade to at least 45.3 or 46.1 once it's released

**Comment by georgelroberts** [comment]
Yes, I've tried syncing multiple times, still nothing. Here are the logs after syncing with the table names removed.

[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.util STARTING: Analyze data for postgres Database 6 'Typewise Browser Extension Analytics'
[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.util STARTING: step 'fingerprint-fields' for postgres Database 6 'Typewise Browser Extension Analytics'
[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.analyze fingerprint-fields Analyzed [*****·············································] 😢   11% Table 162 'public.A'
[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.analyze fingerprint-fields Analyzed [**********········································] 😒   21% Table 135 'public.B'
[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.analyze fingerprint-fields Analyzed [***************···································] 😕   32% Table 91 'public.C'
[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.util FINISHED: step 'fingerprint-fields' for postgres Database 6 'Typewise Browser Extension Analytics' (79.0 ms)
[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.util STARTING: step 'classify-fields' for postgres Database 6 'Typewise Browser Extension Analytics'
[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.analyze classify-fields Analyzed [*********************·····························] 😐   42% Table 175 'public.D'
[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.analyze classify-fields Analyzed [**************************························] 😬   53% Table 161 'public.E'
[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.analyze classify-fields Analyzed [*******************************···················] 😏   63% Table 177 'public.F'
[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.util FINISHED: step 'classify-fields' for postgres Database 6 'Typewise Browser Extension Analytics' (52.7 ms)
[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.util STARTING: step 'classify-tables' for postgres Database 6 'Typewise Browser Extension Analytics'
[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.analyze classify-tables Analyzed [************************************··············] 😋   74% Table 124 'public.G'
[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.analyze classify-tables Analyzed [******************************************········] 😊   84% Table 134 'public.H'
[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.analyze classify-tables Analyzed [***********************************************···] 😍   95% Table 179 'public.I'
[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.util FINISHED: step 'classify-tables' for postgres Database 6 'Typewise Browser Extension Analytics' (70.4 ms)
[d435929f-8523-440c-bed0-9b9fae93b0d0] 2023-04-05T10:37:09+02:00 INFO metabase.sync.util FINISHED: Analyze data for postgres Database 6 'Typewise Browser Extension Analytics' (246.6 ms)

We're running Metabase on Microsoft Azure, taking the image metabase/metabase:latest - does this only get updated on machine restart?

**Comment by paoliniluis** [comment]
Can you check if the tables are hidden in settings->admin->data model?

please change the tag in azure from an immutable tag (latest) to a mutable one (v.x.xx.x)

**Comment by georgelroberts** [comment]
The tables aren't hidden there unfortunately.

I've changed to metabase/metabase:v0.46.0 - I'll restart tonight and update you. Thanks :)

**Comment by georgelroberts** [comment]
Ah the update worked, thanks!
